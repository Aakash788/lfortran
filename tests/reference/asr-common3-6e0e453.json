{
    "basename": "asr-common3-6e0e453",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/common3.f90",
    "infile_hash": "ee4385c2974b7c455c82c16ed901e10b07226305cc917550783f1eab",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-common3-6e0e453.stdout",
    "stdout_hash": "35db8794f745458b986f8245efee9f2bc50321459c25daa2dafe2920",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}