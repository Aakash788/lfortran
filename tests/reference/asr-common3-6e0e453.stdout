(TranslationUnit
    (SymbolTable
        1
        {
            common1:
                (Program
                    (SymbolTable
                        5
                        {
                            1_sample_a:
                                (ExternalSymbol
                                    5
                                    1_sample_a
                                    4 a
                                    sample
                                    []
                                    a
                                    Public
                                ),
                            1_sample_b:
                                (ExternalSymbol
                                    5
                                    1_sample_b
                                    4 b
                                    sample
                                    []
                                    b
                                    Public
                                ),
                            sample:
                                (ExternalSymbol
                                    5
                                    sample
                                    3 sample
                                    file_common_block_sample
                                    []
                                    sample
                                    Public
                                ),
                            struct_instance_sample:
                                (Variable
                                    5
                                    struct_instance_sample
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Struct
                                        5 sample
                                        []
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    common1
                    []
                    [(=
                        (StructInstanceMember
                            (Var 5 struct_instance_sample)
                            5 1_sample_a
                            (Real 4 [])
                            ()
                        )
                        (RealConstant
                            10.000000
                            (Real 4 [])
                        )
                        ()
                    )
                    (=
                        (StructInstanceMember
                            (Var 5 struct_instance_sample)
                            5 1_sample_b
                            (Real 4 [])
                            ()
                        )
                        (RealConstant
                            20.000000
                            (Real 4 [])
                        )
                        ()
                    )
                    (SubroutineCall
                        1 pass
                        ()
                        []
                        ()
                    )]
                ),
            file_common_block_b:
                (Module
                    (SymbolTable
                        8
                        {
                            b:
                                (StructType
                                    (SymbolTable
                                        9
                                        {
                                            d:
                                                (Variable
                                                    9
                                                    d
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            e:
                                                (Variable
                                                    9
                                                    e
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            f:
                                                (Variable
                                                    9
                                                    f
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    b
                                    []
                                    []
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    ()
                                    ()
                                )
                        })
                    file_common_block_b
                    [b]
                    .false.
                    .false.
                ),
            file_common_block_c:
                (Module
                    (SymbolTable
                        6
                        {
                            c:
                                (StructType
                                    (SymbolTable
                                        7
                                        {
                                            c:
                                                (Variable
                                                    7
                                                    c
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            g:
                                                (Variable
                                                    7
                                                    g
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            h:
                                                (Variable
                                                    7
                                                    h
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            j:
                                                (Variable
                                                    7
                                                    j
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    c
                                    []
                                    []
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    ()
                                    ()
                                )
                        })
                    file_common_block_c
                    [c]
                    .false.
                    .false.
                ),
            file_common_block_sample:
                (Module
                    (SymbolTable
                        3
                        {
                            sample:
                                (StructType
                                    (SymbolTable
                                        4
                                        {
                                            a:
                                                (Variable
                                                    4
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    4
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4 [])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    4
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [((IntegerConstant 1 (Integer 4 []))
                                                    (IntegerConstant 5 (Integer 4 [])))])
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    sample
                                    []
                                    []
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    ()
                                    ()
                                )
                        })
                    file_common_block_sample
                    [sample]
                    .false.
                    .false.
                ),
            pass:
                (Function
                    (SymbolTable
                        2
                        {
                            1_sample_a:
                                (ExternalSymbol
                                    2
                                    1_sample_a
                                    4 a
                                    sample
                                    []
                                    a
                                    Public
                                ),
                            1_sample_b:
                                (ExternalSymbol
                                    2
                                    1_sample_b
                                    4 b
                                    sample
                                    []
                                    b
                                    Public
                                ),
                            e:
                                (Variable
                                    2
                                    e
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4 [])
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            sample:
                                (ExternalSymbol
                                    2
                                    sample
                                    3 sample
                                    file_common_block_sample
                                    []
                                    sample
                                    Public
                                ),
                            struct_instance_sample:
                                (Variable
                                    2
                                    struct_instance_sample
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Struct
                                        2 sample
                                        []
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    pass
                    (FunctionType
                        []
                        ()
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        []
                        .false.
                    )
                    []
                    []
                    [(Print
                        ()
                        [(StructInstanceMember
                            (Var 2 struct_instance_sample)
                            2 1_sample_a
                            (Real 4 [])
                            ()
                        )
                        (StructInstanceMember
                            (Var 2 struct_instance_sample)
                            2 1_sample_b
                            (Real 4 [])
                            ()
                        )
                        (Var 2 e)]
                        ()
                        ()
                    )]
                    ()
                    Public
                    .false.
                    .false.
                    ()
                )
        })
    []
)
